test/metabase/lib/filter_test.cljc

```diff
@@ -272,3 +282,27 @@
             (-> query
                 (lib/filter new-filter)
                 lib/filters)))))
+
+(deftest ^:parallel replace-filter-clause-test
+  (testing "Make sure we are able to replace a filter clause using the lib functions for manipulating filters."
+    (let [query           (lib/query-for-table-name meta/metadata-provider "USERS")
+          [first-col]     (lib/filterable-columns query)
+          query           (lib/filter query (lib/filter-clause
+                                             (first (lib/filterable-column-operators first-col))
+                                             first-col
+                                             515))
+          [filter-clause] (lib/filters query)
+          external-op     (lib/external-op filter-clause)]
+      (is (=? {:stages [{:filters [[:= {} [:field {} (meta/id :users :id)] 515]]}]}
+              query))
+      (is (=? [:= {} [:field {} (meta/id :users :id)] 515]
+              filter-clause))
+      (is (=? {:operator "="
+               :lib/type :lib/external-op
+               :args     [[:field {} (meta/id :users :id)]
+                          515]}
+              external-op))
+      (let [external-op' (assoc external-op :operator "!=")
+            query'       (lib/replace-clause query filter-clause external-op')]
+        (is (=? {:stages [{:filters [[:!= {} [:field {} (meta/id :users :id)] 515]]}]}
+                query'))))))
```

Comment:
Also completely unrelated to this issue, but I wrote this test when we were discussing things in a related Slack discussion, doesn't hurt to keep it here